### PR TITLE
Introduce config option to disable statsd metrics

### PR DESCRIPTION
**Pull request by kathap**
This config option allows to send metrics only via prometheus on api vms. The default is true (i.e. enabled). In the default case, metrics are sent via statsd and prometheus.

* Links to any other associated PRs
#3526, #3528, #3521

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

### CODE REVIEW

**Comment by Gerg** [comment]
What do you think about naming the config value `enable_statsd_metrics` instead? That way we don't need to worry about double negatives.

File: lib/cloud_controller/dependency_locator.rb [hunk 1]
```diff
@@ -350,8 +350,13 @@ def statsd_client
       if @dependencies[:statsd_client]
         @dependencies[:statsd_client]
       else
-        Statsd.logger = Steno.logger('statsd.client')
-        register(:statsd_client, Statsd.new(config.get(:statsd_host), config.get(:statsd_port)))
+        config = CloudController::DependencyLocator.instance.config
+        if config.get(:enable_statsd_metrics) == true
```
Comment (philippthun) [inline lib/cloud_controller/dependency_locator.rb:354]: This now evaluates to false for clock and deployment updater, but it shouldn't.

File: spec/unit/lib/cloud_controller/dependency_locator_spec.rb [hunk 2]
```diff
@@ -484,6 +484,41 @@
...
+        and_return(expected_client)
+
+      expect(locator.statsd_client).to eq(expected_client)
+    end
+
+    it 'returns the statsd client for other than api vms' do
+      host = 'test-host'
+      port = 1234
+      TestConfig.context = :deployment_updater
+      TestConfig.override(
```
Comment (philippthun) [inline spec/unit/lib/cloud_controller/dependency_locator_spec.rb:510]: This takes the default values from the `cloud_controller.yml`, i.e. `enable_statsd_metrics: true`. I suggest that you overwrite it with `nil` in this test case.

File: lib/cloud_controller/dependency_locator.rb [hunk 3]
```diff
@@ -350,8 +350,13 @@ def statsd_client
       if @dependencies[:statsd_client]
         @dependencies[:statsd_client]
       else
-        Statsd.logger = Steno.logger('statsd.client')
-        register(:statsd_client, Statsd.new(config.get(:statsd_host), config.get(:statsd_port)))
+        config = CloudController::DependencyLocator.instance.config
```
Comment (philippthun) [inline lib/cloud_controller/dependency_locator.rb:353]: This line is not needed. `config` can be accessed directly.

File: spec/unit/lib/cloud_controller/dependency_locator_spec.rb [hunk 4]
```diff
@@ -484,6 +484,41 @@
 
       expect(locator.statsd_client).to eq(expected_client)
     end
+
+    it 'returns the null statsd client' do
```
Comment (philippthun) [inline spec/unit/lib/cloud_controller/dependency_locator_spec.rb:488]: Somehow the test context is missing, i.e. "when does it return the null statsd client"?

```suggestion
    it 'returns the null statsd client if enable_statsd_metrics is set to false' do
```

File: spec/unit/lib/cloud_controller/dependency_locator_spec.rb [hunk 5]
```diff
@@ -484,6 +484,41 @@
...
+      host = 'test-host'
+      port = 1234
+
+      TestConfig.override(
+        statsd_host: host,
```
Comment (philippthun) [inline spec/unit/lib/cloud_controller/dependency_locator_spec.rb:493]: For this test case host and port are not needed. Removing them makes the test shorter and easier to read.

**Review by philippthun — APPROVED** [review]

**Pull request merged**